lib/loglevel.js

```diff
@@ -189,68 +195,108 @@
       self.methodFactory = factory || defaultMethodFactory;
 
       self.getLevel = function () {
-          if (userLevel != null) {
-            return userLevel;
-          } else if (defaultLevel != null) {
-            return defaultLevel;
-          } else {
-            return inheritedLevel;
+          if (userLevel !== null) {
+              return userLevel;
+          }
+          if (defaultLevel !== null) {
+              return defaultLevel;
           }
+          if (inheritedLevel === null) {
+              inheritedLevel = self.getInheritedLevel();
+          }
+          return inheritedLevel;
       };
 
-      self.setLevel = function (level, persist) {
-          userLevel = normalizeLevel(level);
-          if (persist !== false) {  // defaults to true
-              persistLevelIfPossible(userLevel);
+      self.getInheritedLevel = function() {
```

Comment:
Is it important that this is public? This seems like an implementation detail of `getLevel()`. In keeping with [@pimterry’s earlier comments](https://github.com/pimterry/loglevel/pull/204#issuecomment-2694383418), I think it would be good to keep this private for now. It can always be made public later if there are strong use cases.